internal/credential/vault/repository_ssh_certificate_credential_library_test.go

```diff
@@ -0,0 +1,1294 @@
+			in:      &SSHCertificateCredentialLibrary{},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "invalid-no-store-id",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{},
+			},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "invalid-public-id-set",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:  cs.GetPublicId(),
+					PublicId: "abcd_OOOOOOOOOO",
+				},
+			},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "invalid-no-username",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:   cs.GetPublicId(),
+					VaultPath: "/some/path",
+				},
+			},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "invalid-no-vault-path",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:   cs.GetPublicId(),
+					VaultPath: "/some/path",
+				},
+			},
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "valid-no-options",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:   cs.GetPublicId(),
+					KeyType:   "ed25519",
+					VaultPath: "/some/path",
+					Username:  "admin",
+				},
+			},
+			want: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:   cs.GetPublicId(),
+					KeyType:   "ed25519",
+					VaultPath: "/some/path",
+					Username:  "admin",
+				},
+			},
+		},
+		{
+			name: "valid-with-name",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:   cs.GetPublicId(),
+					KeyType:   "ed25519",
+					Name:      "test-name-repo",
+					VaultPath: "/some/path",
+					Username:  "admin",
+				},
+			},
+			want: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:   cs.GetPublicId(),
+					KeyType:   "ed25519",
+					Name:      "test-name-repo",
+					VaultPath: "/some/path",
+					Username:  "admin",
+				},
+			},
+		},
+		{
+			name: "valid-with-description",
+			in: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:     cs.GetPublicId(),
+					KeyType:     "ed25519",
+					Description: "test-description-repo",
+					VaultPath:   "/some/path",
+					Username:    "admin",
+				},
+			},
+			want: &SSHCertificateCredentialLibrary{
+				SSHCertificateCredentialLibrary: &store.SSHCertificateCredentialLibrary{
+					StoreId:     cs.GetPublicId(),
+					KeyType:     "ed25519",
+					Description: "test-description-repo",
+					VaultPath:   "/some/path",
+					Username:    "admin",
+				},
+			},
```

Comment:
Can you add some additional test cases, specifically around:
- `KeyType` and `KeyBits` both valid and invalid combinations
- CriticalOptions
- Extensions